tools/nuget/README.md

```diff
@@ -0,0 +1,44 @@
+# NuGet package creation process
+
+This directory contains NuGet specification files (.nuspec).
+
+Creating nuget packages:
+
+1. Set `PackageVersion` environment variable that defines the package version.
+
+```console
+set PackageVersion=0.2.0
+```
+
+If environment variable is not set, then the version is automatically obtained from git tag:
+
+```console
+git describe --tags
+```
+
+2. Run `tools/build-nuget.cmd` script to create the package(s). Packages are copied to `.\packages` directory.
+
+3. Push selected package to nuget feed.
+
+Learn more about NuGet packages [here](https://docs.microsoft.com/en-us/nuget/guides/native-packages).
+
+## Proposed NuGet package flavors
+
+This section is a `DRAFT` - Work-in-Progress.
+
+The following packages MAY be created and published at nuget.org :
+
+| Package Name                 | Desription                                                  | Notes |
+|------------------------------|-------------------------------------------------------------|-------|
+| OpenTelemetry.Cpp.Source     | Complete recursive source code snapshot with dependencies.  |       |
```

Comment:
@cijothomas need to coordinate the package name here.

Reply:
@maxgolov FYI https://www.nuget.org/profiles/OpenTelemetry

Reply:
My main concern here is that some C++ projects name their packages all lower-case, e.g. `boost` or `cpprestsdk` . Whereas I believe for OpenTelemetry, we should probably stick to common CamelCase naming convention. I isolated all C++ packages under `OpenTelemetry.Cpp.*`

Reply:
I think we have freedom here to call it `opentelemetry.cpp` or `otel.cpp` or whatever name that sounds native to C++ developers.

Reply:
@cijothomas @reyang - I think if we use `OpenTelemetry` prefix, then it's gonna be easier to find the package. That way .NET developers already using the package would see that there's been a package created for native code as well, and spread the word.